gtsam/base/Matrix.cpp

```diff
@@ -30,6 +30,7 @@
 #include <fstream>
 #include <limits>
 #include <iostream>
+#include <iterator>
```

Comment:
Can you please explain the need for this header and other new headers? Seems strange it is needed now in such a core file.

Reply:
Our system failed to compile without the header.